Version bump to 0.0.3 and fix NPM publish workflow

## Changes

This issue tracks the following changes:

1. **Version Update**: Bumped package version from 0.0.2 to 0.0.3 in package.json
2. **CI/CD Fix**: Added \ flag to the NPM publish command in the CD workflow

## Details

### package.json
- Updated version field from \0.0.2\ to \0.0.3\

### .github/workflows/cd.yml  
- Modified NPM publish command from \undefined
 ERR_PNPM_RECURSIVE_EXEC_FIRST_FAIL  Command failed with EACCES: publish\
spawn publish\ EACCES to \
- This change prevents git-related publish failures in the CI environment

## Impact
- Prepares for new release version 0.0.3
- Ensures NPM publishing works correctly in GitHub Actions workflow

--- a/.github/workflows/cd.yml
+++ b/.github/workflows/cd.yml
@@ -40,6 +40,6 @@ jobs:
       run: pnpm run build
     
     - name: Publish to NPM
-      run: pnpm publish
+      run: pnpm publish --no-git-checks
       env:
         NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "typescript-eslint-extended-switch-exhaustiveness",
-  "version": "0.0.2",
+  "version": "0.0.3",
   "description": "ESLint plugin that enforces 'throw new Error(`${value satisfies never} ...`)' in default cases for union types",
   "type": "module",
   "main": "dist/index.js",

